src/endian/stream.hpp

```diff
@@ -15,16 +15,13 @@ class stream
 {
 public:
 
-    /// Creates an endian stream on top of a pre-allocated buffer of the
-    /// specified size.
+    /// Creates an endian stream used to track a buffer of the specified size.
     ///
-    /// @param buffer a pointer to the buffer
     /// @param size the size of the buffer in bytes
     stream(uint32_t size) :
-        m_size(size),
-        m_position(0)
+        m_size(size)
     {
-        assert(m_size > 0);
+        assert(size > 0 && "Tracking buffer of size zero is not valid");
```

Comment:
I fundamentally disagree, I think asserts should be used to check for the validity of input parameters, not the correctness of the implementation. From the users point of view he gets told that some member variable have an invalid value - which is not really his/her problem... do you see what I mean?